gdx/src/com/badlogic/gdx/graphics/g2d/PixmapPacker.java

```diff
@@ -326,66 +322,61 @@ public synchronized void dispose () {
+	 */
+	public synchronized void updateTextureAtlas(TextureAtlas atlas, TextureFilter minFilter, TextureFilter magFilter,
 		boolean useMipMaps) {
 		for (Page page : pages) {
-			if (page.texture == null) {
-				if (page.rects.size != 0 && page.addedRects.size > 0) {
-					page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
-						@Override
-						public void dispose () {
-							super.dispose();
-							getTextureData().consumePixmap().dispose();
-						}
-					};
-					page.texture.setFilter(minFilter, magFilter);
-
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
-							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
-				}
-			} else {
-				if (page.addedRects.size > 0) {
+			if (page.addedRects.size > 0) {
+				if (page.texture == null) {
+					generatePageTexture(page, minFilter, magFilter, useMipMaps);
+				} else {
 					page.texture.load(page.texture.getTextureData());
-					for (String name : page.addedRects) {
-						Rectangle rect = page.rects.get(name);
-						TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
+				}
+				for (String name : page.addedRects) {
+					Rectangle rect = page.rects.get(name);
+					TextureRegion region = new TextureRegion(page.texture, (int)rect.x, (int)rect.y, (int)rect.width,
 							(int)rect.height);
-						atlas.addRegion(name, region);
-					}
-					page.addedRects.clear();
+					atlas.addRegion(name, region);
 				}
+				page.addedRects.clear();
+			}
+		}
+	}
+
+	private void generatePageTexture(final Page page, TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		page.texture = new Texture(new PixmapTextureData(page.image, page.image.getFormat(), useMipMaps, false, true)) {
+			@Override
+			public void dispose() {
+				super.dispose();
+				getTextureData().consumePixmap().dispose();
+			}
+		};
+		page.texture.setFilter(minFilter, magFilter);
+	}
+
+	/** Return an ordered array of texture regions corresponding with the pages of the packer, creating new textures
+	 * if necessary.  The same array will be returned from successive calls to this method, possibly modified with
+	 * additional texture regions if more pages have been added since the last call.  Once this method has been called,
+	 * you must dispose of the resulting texture regions (or equivalently, any atlas generated or updated by this packer)
+	 * yourself - disposing the PixmapPacker will no longer dispose these textures or their backing pixmaps.
+	 */
+	public Array<TextureRegion> getTextureRegions(TextureFilter minFilter, TextureFilter magFilter, boolean useMipMaps) {
+		while (textureRegions.size < pages.size) {
+			if (disposed) return null;

```

Comment:
I agree it's getting into undocumented feature territory.  My reasoning was that the existing checks of `disposed` didn't forbid all access to the Packer, just things that relied on the resources which had been disposed, so I was following suite.

Building a new TextureRegion requires access to the pixmaps, so it can't be done after disposing the packer, but the cached array of TextureRegions is still valid.

I'm happy to move the `disposed` check out of the while loop so that calling `getTextureRegion` always returns null on a disposed PixmapPacker if you like.